### PR TITLE
Allow for moving points in dirac kernels

### PR DESCRIPTION
**Pull request by GiudGiud**
if especially asked for in the input
refs #24399

The alternative is to write your point source in the expected format. Seems less flexible
https://mooseframework.inl.gov/source/dirackernels/ReporterTimePointSource.html

### CODE REVIEW

**Review by cpgr — APPROVED** [review]
Thanks for the quick fix.

Super minor point, but should it be `clearPointsCache` rather than `clearPointsCaches`?

**Comment by GiudGiud** [comment]
There's two caches

**Comment by GiudGiud** [comment]
@lindsayad we can review / merge this now or wait for my test which involve a complex functor reporter. Your call

**Comment by cpgr** [comment]
> There's two caches

Haha! Ignore my comment then!

File: framework/src/dirackernels/DiracKernelBase.C [hunk 1]
```diff
@@ -279,7 +285,8 @@ DiracKernelBase::addPointWithValidId(Point p, unsigned id)
                  " already exists with ID: ",
                  id,
                  " and does not match point ",
-                 p);
+                 p,
+                 "If Dirac sources are moving, please set 'allow_moving_sources' to true");
```
Comment (GiudGiud) [inline framework/src/dirackernels/DiracKernelBase.C:289]:
```suggestion
                 "\n. If Dirac sources are moving, please set 'allow_moving_sources' to true");
```

**Comment by GiudGiud** [comment]
The functor reporter is turning out to be difficult, due to some interaction between vecStore and reporters of vectors of points.

**Review by lindsayad — APPROVED** [review]
This looks good to me. Don't know whether you want to commit your suggestion there

**Pull request merged**